great_expectations/expectations/expectation.py

```diff
@@ -104,6 +104,17 @@
 P = ParamSpec("P")
 T = TypeVar("T", List[RenderedStringTemplateContent], RenderedAtomicContent)
 
+try:
```

Comment:
Not putting this in the compat layer because it's non-functioning (but that's okay for our needs here)

Reply:
Do we need to wrap this in a try/except block? It looks like this appears in typing-extensions 3.11 and we require pydantic>=3.9.2 and [3.9.2 pins typing-extensions 4.1.0](https://github.com/pydantic/pydantic/blob/v1.9.2/requirements.txt)

Does that mean no autocomplete for people with the old typing-extensions version? If so, we should make sure that gets in our docs somewhere.

Reply:
`non-blocking`
@tyler-hoffman I don't follow why we wouldn't put this in the `typing-extensions` compat layer.
The existing `@override` decorator that's in there isn't functional (at runtime) either.